hadoop-hdds/managed-rocksdb/src/main/java/org/apache/hadoop/hdds/utils/db/managed/ManagedRocksObjectUtils.java

```diff
@@ -41,27 +42,24 @@ private ManagedRocksObjectUtils() {
 
   public static final Logger LOG =
       LoggerFactory.getLogger(ManagedRocksObjectUtils.class);
+
   private static final Duration POLL_DELAY_DURATION = Duration.ZERO;
   private static final Duration POLL_INTERVAL_DURATION = Duration.ofMillis(100);
 
-  public static void assertClosed(ManagedObject<?> object) {
-    assertClosed(object.get(), object.getStackTrace());
-  }
 
-  static void assertClosed(RocksObject rocksObject, String stackTrace) {
+  private static final LeakDetector LEAK_DETECTOR = new LeakDetector("ManagedRocksObject");
+
+  static LeakTracker track(AutoCloseable object, @Nullable StackTraceElement[] stackTrace) {
     ManagedRocksObjectMetrics.INSTANCE.increaseManagedObject();
-    if (rocksObject.isOwningHandle()) {
-      reportLeak(rocksObject, stackTrace);
-    }
+    final String name = object.getClass().getSimpleName();
```

Comment:
Use `object.toString() instead`.

Reply:
@duongkame , any comments on this?

Reply:
I don't think we need anything specific in `toString()`. Plus, `toString` gives an uncertain performance.

Reply:
I fine to use `Class.getSimpleName()` although it is known to have bad performance; see https://stackoverflow.com/questions/17369304/why-is-class-getsimplename-not-cached
It seems minor.